### PR TITLE
Project actualized according to recommendations and implementation de…

### PR DESCRIPTION
**Pull request by EvgenyBelichkov**
Hello Andrew!
I add some code to the project. Can I ask you to check it before the final exam?
In this code I add:
1) tests to the project;
2) I changed the connection to the database as you recommended. Now I used ORM as should be;
3) Add pagination and one simple front-end page.
Would be glad to get any recommendations.
Thank you in advance.

### CODE REVIEW

File: EPAM_PROJECT/checker/views.py [hunk 1]
```diff
@@ -16,119 +17,95 @@ def contacts(request):
...
-):  # noqa
-    with connection(database_name) as cursor:
-        cursor.execute(
-            f"INSERT INTO {table_name}(city_name, current_temperature, "
-            f"feels_like_temperature, wind_speed, time_created ) "
-            f"VALUES(?, ?, ?, ?, ?)",
-            (city_name, temp, f_l_temp, wind_speed, datetime.datetime.now()),
-        )
+def insert_row(city_name, temp, f_l_temp, wind_speed):
+    insert_information = WeatherStory(
```
Comment (Cjkjvfnby) [inline EPAM_PROJECT/checker/views.py]: You can inline this variable.

File: EPAM_PROJECT/checker/views.py [hunk 2]
```diff
@@ -16,119 +17,95 @@ def contacts(request):
...
+    updated_row.wind_speed = wind_speed
+    updated_row.time_created = datetime.datetime.now()
+    updated_row.save()
 
 
-def api_1(name_city):
-    api_url = "https://api.weatherbit.io/v2.0/current"
-    params = {
-        "city": name_city,
+def getting_weather_from_two_different_api(requested_city):
```
Comment (Cjkjvfnby) [inline EPAM_PROJECT/checker/views.py]: The idea behind splitting code to small function is that when you need to change it you do it with small impact.  

Basically, this function could look like this at first iteration:

```python
sources = [
   source1, source2
]
return merge(*[x() for x in sources])
```

Later you could replace it with populating sources from settings.

File: EPAM_PROJECT/checker/static/checker/css/style.css [hunk 3]
```diff
@@ -122,6 +121,14 @@
...
+    display: flex;
+    align-items: center;
+    flex-direction: row;
+    color: white;
+    font-family: Roboto;
+}
+
 .widthresult {
     width: 45%;
 }
```
Comment (Cjkjvfnby) [inline EPAM_PROJECT/checker/static/checker/css/style.css]: image

Please pay attention to such things.  You can fix It manually or google how to configure your IDE.

File: EPAM_PROJECT/checker/templates/checker/cities_on_page.html [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+{% block content %}
+
+    <div class="container background2">
+        <div class="box7 mission1">
+            <div style="font-weight: 700; width: 70%" class="box8"> Previously requested cities </div>
+            {% for city in page_cities %}
+            <div style="width: 70%" class="box8"> {{ city }} </div>
+            {% endfor %}
+        </div>
+            <p class="box7">
```
Comment (Cjkjvfnby) [inline EPAM_PROJECT/checker/templates/checker/cities_on_page.html]: Strange indent.

**Review by Cjkjvfnby — COMMENTED** [review]
What is about changing time without changing code?

File: EPAM_PROJECT/checker/views.py (hunk 1 above)
Comment (EvgenyBelichkov) [reply EPAM_PROJECT/checker/views.py]: Variable realized inline

File: EPAM_PROJECT/checker/views.py (hunk 2 above)
Comment (EvgenyBelichkov) [reply EPAM_PROJECT/checker/views.py]: I did as you advise but using the dictionary.

File: EPAM_PROJECT/checker/static/checker/css/style.css (hunk 3 above)
Comment (EvgenyBelichkov) [reply EPAM_PROJECT/checker/static/checker/css/style.css]: IDE configurated.

File: EPAM_PROJECT/checker/templates/checker/cities_on_page.html (hunk 4 above)
Comment (EvgenyBelichkov) [reply EPAM_PROJECT/checker/templates/checker/cities_on_page.html]: Indent corrected.

File: EPAM_PROJECT/EPAM_PROJECT/settings/local.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+from .settings import *  # noqa
+
+# Time duration for weather checker application
+CURRENT_DURATION = datetime.timedelta(seconds=10)
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/3.1/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = "$)bxhkmx(2fv92*%c)^%k^v#o!3=_g_am%*@9rg651!z)&6kjg"
```
Comment (Cjkjvfnby) [inline EPAM_PROJECT/EPAM_PROJECT/settings/local.py]: This looks like a secret, it's better not to make it look like a secret.

File: EPAM_PROJECT/EPAM_PROJECT/settings/prod.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+from .settings import *  # noqa
+
+# Time duration for weather checker application
+CURRENT_DURATION = datetime.timedelta(seconds=60)
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/3.1/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = "$)bxhkmx(2fv92*%c)^%k^v#o!3=_g_am%*@9rg651!z)&6kjg"
```
Comment (Cjkjvfnby) [inline EPAM_PROJECT/EPAM_PROJECT/settings/prod.py]: Why do you ignore the comment above?

File: EPAM_PROJECT/checker/tests/test_views.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        "current_temperature": 30,
+        "feels_like_temperature": 35,
+        "wind_speed": 7,
+        "time_created": datetime.datetime(2021, 5, 19),
+    }
+
+    def set_up(self):
+        self.client = Client()
+
+    def test_counting_average_mean_of_weather_example_1(self):
```
Comment (Cjkjvfnby) [inline EPAM_PROJECT/checker/tests/test_views.py:31]: This is a unittest inside the integration test class.  I'd prefer to move it to a separate file.

**Pull request merged**